### PR TITLE
feat: modal method rtl

### PR DESCRIPTION
**Pull request by xrkffgg**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/24566

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add `Modal.method()` RTL supports only use hooks.       |
| 🇨🇳 Chinese | 增加 `Modal.method()` RTL 模式支持，仅限 hooks 用法。          |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/modal/ConfirmDialog.tsx [hunk 1]
```diff
@@ -50,9 +51,12 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
   const transitionName = props.transitionName || 'zoom';
   const maskTransitionName = props.maskTransitionName || 'fade';
 
+  const { direction } = React.useContext(ConfigContext);
```
Comment (xrkffgg) [inline components/modal/ConfirmDialog.tsx:54]: help help help

这个地方获 不到 `direction` 啊

Comment (zombieJ) [reply components/modal/ConfirmDialog.tsx:54]: 参考 Modal or Notification 的 hooks 用法，需要有个上下文环境才行。

**Comment by xrkffgg** [comment]
赶得上这班车吗？

**Comment by afc163** [comment]
赶不上，没必要赶。

File: components/modal/index.zh-CN.md [hunk 2]
```diff
@@ -48,7 +48,9 @@ cover: https://gw.alipayobjects.com/zos/alicdn/3StSdUlSH/Modal.svg
 
 #### 注意
 
-> `<Modal />` 默认关闭后状态不会自动清空, 如果希望每次打开都是新内容，请设置 `destroyOnClose`。
+> - `<Modal />` 默认关闭后状态不会自动清空, 如果希望每次打开都是新内容，请设置 `destroyOnClose`。
+>
+> - `Modal.method()` RTL 模式仅支持 hooks 写法。
```
Comment (xrkffgg) [inline components/modal/index.zh-CN.md:53]:
```suggestion
> - `Modal.method()` RTL 模式仅支持 hooks 用法。
```

**Comment by xrkffgg** [comment]
rebase 不好用了？

**Comment by yoyo837** [comment]
可能有冲突，不成功会没反应。

**Comment by yoyo837** [comment]
手动rebase

**Comment by xrkffgg** [comment]
木有冲突啊，手动很顺。

**Comment by yoyo837** [comment]
那小弟就不知道为啥了

**Comment by xrkffgg** [comment]
PTAL

**Comment by crazyair** [comment]
image
现在好像没问题了，是不是可以去掉这个描述

**Pull request merged**